src/modules/users/test/users.controller.spec.ts

```diff
@@ -108,4 +123,59 @@ describe('UsersController', () => {
       expect(controller.deleteUser('1')).resolves.toBeUndefined();
     });
   });
+
+  describe('wishlistAccommodation', () => {
+    it('should add accommodation to wishlist if not already in', async () => {
+      const userId = mockUsers[0].id;
+      const accommodationId = 'accommodation-2'; // New accommodation ID
+      const mockRequest = { user: { sub: userId } }; // Mocking the request object
+
+      // Mock the service method to resolve successfully
+      jest.spyOn(service, 'wishlistAccommodation').mockResolvedValue(undefined);
+
+      await expect(
+        controller.wishlistAccommodation(accommodationId, mockRequest),
+      ).resolves.not.toThrow();
+
+      // Check if the service method was called with the correct arguments
+      expect(service.wishlistAccommodation).toHaveBeenCalledWith(
+        userId,
+        accommodationId,
+      );
+    });
+
+    it('should remove accommodation from wishlist if already in', async () => {
+      const userId = mockUsers[0].id;
+      const accommodationId = 'accommodation-1'; // Already in the wishlist
+      const mockRequest = { user: { sub: userId } };
+
+      // Mock the service method to resolve successfully
+      jest.spyOn(service, 'wishlistAccommodation').mockResolvedValue(undefined);
+
+      await expect(
+        controller.wishlistAccommodation(accommodationId, mockRequest),
+      ).resolves.not.toThrow();
+
+      // Check if the service method was called with the correct arguments
+      expect(service.wishlistAccommodation).toHaveBeenCalledWith(
+        userId,
+        accommodationId,
+      );
+    });
+
+    it('should throw NotFoundException if user or accommodation not found', async () => {
+      const userId = 'non-existing-user-id'; // Invalid user ID
+      const accommodationId = 'accommodation-1'; // Valid accommodation ID
+      const mockRequest = { user: { sub: userId } };
+
+      // Simulate the service throwing an error
+      jest
```

Comment:
it is better to handle service methods mocking when mocking the module. you can use mockImplementation from jest to handle almost any cases (returning values, throwing errors etc...).